_episodes/05-creating-container-images.md

```diff
@@ -117,8 +117,22 @@ Let's break this file down:
 - The first line, `FROM`, indicates which container we're starting with.
 - The next two lines `RUN`, will indicate installation commands we want to run. These
 are the same commands that we used interactively above.
-- The last line, `CMD` indicates the default command we want the container to run,
-if no other command is provided.
+- The last line, `CMD`, indicates the default command we want the
+container to run, if no other command is provided. It is recommended
+to provide `CMD` in *exec-form*, that is a JSON-array which contains
+as first element the executable, optionally followed by its
+arguments. For example: `CMD ["ls", "-lF", "--color", "/etc"]`
+
+> ## *shell-form* and *exec-form* for CMD
+> Another way to specify the `CMD` is the *shell-form*. Her you just
+> type the command as you would call it from the command line. Docker
+> then silently runs this command in the image's standard shell.  `CMD
+> cat /etc/passwd` is equivalent to `CMD
+> ["/bin/sh", "-c", "cat /etc/passwd"]`. We recommend to prefer the
```

Comment:
No, it won't work this way (tried it). I can only explain this on an intuitive (hopefully) level: When you give the following command on the command line (interactive shell) it won't work:
`/bin/sh -c cat /etc/passwd`
That is because the interactive shell parses your command into tokens (everything separated by whitespace), calls the first token as executable (`/bin/sh`) with all of the remaining tokens as individual arguments to that executable. However, `sh` expects after `-c` just **one** argument, namely one **command_string** (which it then subsequently parses by itself). So to make that work on the command-line you have to say:
`/bin/sh -c "cat /etc/passwd"`
 which (I hope that is correct) tells your interactive shell that it is supposed to treat `"cat /etc/passwd"` as **one** token.
Now my mental model how to get the exec-form is "to do the splitting into tokens by yourself", instead of having it done by the interactive (or implicitly called) shell.
Hope that makes sense. In any case, I don't see a way to do this explaining in the lesson. Maybe put it into the instructor-notes to be prepared for questions?